joomla/joomla-cms
# [4.0] Move namespacing be explicit

### Summary of Changes
Moves namespaces to be explicit rather than magic

@laoneo this is your chance to have the argument about this finally ;)

### Testing Instructions
Install Joomla. ENSURE YOU HAVE DELETED `libraries/autoload_psr4` before testing if installing it into an existing 4.x install

### Expected result
Everything still works

TODO: Add support for plugins - currently this only handles components and modules

### Documentation Changes Required
Document that extensions to do autoloading must declare their base namespace in their XML files
